dot/node.go

```diff
@@ -144,13 +145,13 @@ func (*nodeBuilder) initNode(cfg *Config) error {
 	}
 
 	// create trie from genesis
-	t, err := genesis.NewTrieFromGenesis(gen)
+	t, err := wasmer.NewTrieFromGenesis(*gen)
 	if err != nil {
 		return fmt.Errorf("failed to create trie from genesis: %w", err)
 	}
 
 	// create genesis block from trie
-	header, err := genesis.NewGenesisBlockFromTrie(t)
+	header, err := t.GenesisBlock()
```

Comment:
this is moved as a method on the trie since it doesn't add dependencies in lib/trie, but allows to remove the lib/trie dependency in lib/genesis